dowhy/gcm/uncertainty.py

```diff
@@ -0,0 +1,78 @@
+"""Functions to estimate uncertainties such as entropy, KL divergence etc.
+
+Functions in this module should be considered experimental, meaning there might be breaking API changes in the future.
+"""
+
+import numpy as np
+from numpy.dual import det
+from scipy.special import digamma
+from scipy.stats import entropy
+from sklearn.neighbors import NearestNeighbors
+
+from dowhy.gcm.constant import EPS
+from dowhy.gcm.util.general import shape_into_2d
+
+
+# TODO: Add method for auto select a bin_width/width based on the data. Make sure that the auto selection method is
+#  theoretically sound, i.e. make entropy results from different data comparable.
+def estimate_entropy_discretization(X: np.ndarray, bin_width: float = 1) -> float:
```

Comment:
There is no method yet that makes directly use of the fact that data is discrete. I guess a simple modification of ```estimate_entropy_discretization``` could do this. Definitely something to add. Generally, I would expect that ```estimate_entropy_kmeans``` would give very similar results.

Reply:
Got it, will be good to add it since datasets often have categorical variables too. You may either choose to add it now or perhaps add it as a "good first issue" on the Issues page so it can be taken up later. With that in mind, I'm approving this PR.

Reply:
I added an implementation for discrete data to the commit.